pkg/datapath/linux/config/config.go

```diff
@@ -549,6 +555,74 @@ func (h *HeaderfileWriter) WriteNodeConfig(w io.Writer, cfg *datapath.LocalNodeC
 	return fw.Flush()
 }
 
+// vlanFilterMacros generates VLAN_FILTER macros which
+// are written to node_config.h
+func vlanFilterMacros() (string, error) {
+	devices := make(map[int]bool)
+	for _, device := range option.Config.Devices {
+		ifindex, err := link.GetIfIndex(device)
+		if err != nil {
+			return "", err
+		}
+		devices[int(ifindex)] = true
+	}
+
+	allowedVlans := make(map[int]bool)
+	for _, vlanId := range option.Config.VLANBPFBypass {
+		allowedVlans[vlanId] = true
+	}
+
+	// allow all vlan id's
+	if allowedVlans[0] {
+		return "return true", nil
+	}
+
+	vlansByIfIndex := make(map[int][]int)
+
+	links, err := netlink.LinkList()
+	if err != nil {
+		return "", err
+	}
+
+	for _, l := range links {
+		vlan, ok := l.(*netlink.Vlan)
+		// if it's vlan device and we're controlling vlan main device
+		// and either all vlans are allowed, or we're controlling vlan device or vlan is explicitly allowed
+		if ok && devices[vlan.ParentIndex] && (devices[vlan.Index] || allowedVlans[vlan.VlanId]) {
+			vlansByIfIndex[vlan.ParentIndex] = append(vlansByIfIndex[vlan.ParentIndex], vlan.VlanId)
```

Comment:
This appending of vlanIDs that may not be ordered introduced a new unit test flake #17104.

Reply:
@kvaster would you be able to look into this? Usually the solution is just to sort the slice once it's been assembled but before it is used to generate code.

Reply:
Never mind, looks like @jrajahalme beat you to it :) Feel free to review though: https://github.com/cilium/cilium/pull/17105

Reply:
Was away... Seems problem was already solved... I will make one more PR soon as discussed - docs update.